src/main/java/com/celements/model/classes/ClassDefinition.java

```diff
@@ -0,0 +1,33 @@
+package com.celements.model.classes;
+
+import java.util.List;
+
+import org.xwiki.component.annotation.ComponentRole;
+import org.xwiki.model.reference.DocumentReference;
+
+import com.celements.model.classes.fields.ClassField;
+
+@ComponentRole
+public interface ClassDefinition {
+
+  /**
+   * @return the name of the component and class definition, used for blacklisting
+   */
+  public String getName();
+
+  /**
+   * @return the document reference on which the class is defined
+   */
+  public DocumentReference getClassRef();
+
+  /**
+   * @return true if the class is mapped internally (hibernate mapping)
+   */
+  public boolean isInternalMapping();
+
+  /**
+   * @return a list of all fields defining this class
+   */
+  public abstract List<ClassField<?>> getFields();
```

Comment:
please do not use "abstract" in an interface. Interface is per definition "abstract"